### PR TITLE
#2735 Fixed order in which branches are pushed to upstream

### LINKED ISSUE #2735: configuration-service: HEAD branch not properly set
When a Keptn project is created w/o upstream and afterward an empty GIT repo is set as upstream, the `HEAD branch` is not properly set. In my case it points to `dev`:
```
git remote show origin
* remote origin
  Fetch URL: https://keptn-nightly-user:4f0a9b9e9122ef19c093de2226556d4090fec4ec@github.com/my-keptn-tests/sockshop2.git
  Push  URL: https://keptn-nightly-user:4f0a9b9e9122ef19c093de2226556d4090fec4ec@github.com/my-keptn-tests/sockshop2.git
  HEAD branch: dev
  Remote branches:
    dev        new (next fetch will store in remotes/origin)
    hardening  new (next fetch will store in remotes/origin)
    master     new (next fetch will store in remotes/origin)
    production new (next fetch will store in remotes/origin)
  Local refs configured for 'git push':
    dev        pushes to dev        (up to date)
    hardening  pushes to hardening  (up to date)
    master     pushes to master     (up to date)
    production pushes to production (up to date)
```

This is because the first branch we use in `git push --set-upstream URI branch` gets the HEAD branch and the branches are ordered alphabetically.

**Impact:**
When someone queries **project** resources contained in the main/master branch, they are not found because the function `GetDefaultBranch` returns the wrongly set HEAD branch, i.e. see
https://github.com/keptn/keptn/blob/9cd0b8e3eb92b83ab734bb57c170de67d4f077f4/configuration-service/common/git.go#L314

**Possible solution:**
- Change the order how the branches are pushed, i.e. first push the desired HEAD branch (master or main). The code can be found here
 https://github.com/keptn/keptn/blob/9cd0b8e3eb92b83ab734bb57c170de67d4f077f4/configuration-service/common/git.go#L135

### PR DESCRIPTION
**Pull request by bacherfl**
Closes #2735 

Signed-off-by: Florian Bacher <florian.bacher@dynatrace.com>

### CODE REVIEW

File: configuration-service/deploy/service.yaml [hunk 1]
```diff
@@ -60,7 +60,7 @@ spec:
             memory: "256Mi"
             cpu: "500m"
           limits:
-            memory: "256Mi"
+            memory: "1024Mi"
```
Comment (christian-kreuzberger-dtx) [inline configuration-service/deploy/service.yaml:63]: The higher limit is only needed for debugging, right?

**Review by christian-kreuzberger-dtx — DISMISSED** [review]
Looks good to me.
@agrimmer can you validate if the changes made by Florian are performing as expected?

File: configuration-service/deploy/service.yaml (hunk 1 above)
Comment (bacherfl) [reply configuration-service/deploy/service.yaml:63]: yes - it's not directly related to the issue but with the previous setting i got an OOM error when trying to attach the debugger

**Review by johannes-b — APPROVED** [review]
lgtm

**Pull request merged**